packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -413,13 +413,15 @@ async function ensureAvailableUsers(
         recurringDatesInfo?.currentRecurringIndex === 0 &&
         recurringDatesInfo.allRecurringDates
       ) {
-        const allBookingDates = recurringDatesInfo.allRecurringDates.map((strDate) => new Date(strDate));
+        const firstTwoBookingDates = recurringDatesInfo.allRecurringDates
+          .slice(0, 2)
+          .map((strDate) => new Date(strDate));
         // Go through each date for the recurring event and check if each one's availability
         // DONE: Decreased computational complexity from O(2^n) to O(n) by refactoring this loop to stop
         // running at the first unavailable time.
         let i = 0;
-        while (!foundConflict && i < allBookingDates.length) {
```

Comment:
I don't think this would actually work.

Here is why:
`bufferedBusyTimes` contains busy times only for the DateRange for which `handleNewBooking` was called and we don't pass it the entire 12 weeks date range(for a 12 occurence recurring event), we actually call it 12 times with 12 different timeslots.

So, I believe this code is dead code as of now, maybe due to some refactor.